pkg/supportbundle/spec.go

```diff
@@ -106,52 +106,69 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
 		}
 	}
 
-	builtBundle, err = injectDefaults(app, builtBundle, opts, namespacesToCollect, namespacesToAnalyze)
+	// split the default kotsadm support bundle into multiple support bundles
+	builtBundles, err := injectDefaults(app, builtBundle, opts, namespacesToCollect, namespacesToAnalyze)
+
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to inject defaults")
 	}
 
+	for key, builtBundle := range builtBundles {
+		err := createSupportBundleSpecSecret(app, sequence, kotsKinds, key, builtBundle, opts, clientset)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create support bundle secret")
+		}
+	}
+
+	// Include discovered support bundle specs and multiple kotsadm support bundle specs. Perform this action here so
+	// as not to add discovered specs to the default support bundle spec secret.c
+	// use cluster specific support bundle spec to add Spec.AfterCollection as default
+	return addDiscoveredSpecs(builtBundles[kotstypes.ClusterSpecificSupportBundleSpecKey], app, clientset), nil
```

Comment:
This results in the generated bundle missing all of the collectors/analyzers from the "default" and "vendor" specs since those are filtered out by the `^kotsadm-.*-supportbundle$` regex in `addDiscoveredSpecs(...)`.

We could maybe do something like this (pseudo-code):
```
defaultBundle := builtBundles[kotstypes.DefaultSupportBundleSpecKey]
vendorBundle := builtBundles[kotstypes.VendorSpecificSupportBundleSpecKey]
clusterSpecificBundle := builtBundles[kotstypes.ClusterSpecificSupportBundleSpecKey]

// add discovered collectors and analyzers to cluster-specific
clusterSpecificBundle = addDiscoveredSpecs(clusterSpecificBundle, app, clientset)

// merge all collectors and analyzers (need to make sure AfterCollection is maintained)
mergedBundle := mergeBundles(defaultBundle, vendorBundle, clusterSpecificBundle)

return mergedBundle
```

Reply:
After some discussion, this was not the reason why the bundle was empty.  It was failing to collect with an error:
```
"error collecting support bundle ID from URI: 2mknjabqcz7sx9rorg9poukpg0v: could not bundle from URI: failed to load collector spec: failed to get spec from secret: failed to get secret: secrets \"kotsadm-craig-helm-supportbundle\" not found"
```

Reply:
Hey @cbodonnell, I have added  mergedBundle to combine three separated specs into one and saved as origin one for backward compatible. Also, clusterSpecificBundle has been changed to use discovery function